code/__HELPERS/mobs.dm

```diff
@@ -274,32 +274,41 @@ Proc for attack log creation, because really why not
 This is always put in the attack log.
 */
 
-/proc/add_attack_logs(mob/user, mob/target, what_done, custom_level)
+/proc/add_attack_logs(atom/user, target, what_done, custom_level)
 	if(islist(target)) // Multi-victim adding
 		var/list/targets = target
 		for(var/mob/M in targets)
 			add_attack_logs(user, M, what_done, custom_level)
 		return
 
 	var/user_str = key_name_log(user) + COORD(user)
-	var/target_str = key_name_log(target) + COORD(target)
-
-	if(istype(user))
-		user.create_attack_log("<font color='red'>Attacked [target_str]: [what_done]</font>")
-	if(istype(target))
-		target.create_attack_log("<font color='orange'>Attacked by [user_str]: [what_done]</font>")
+	var/target_str
+	if(isatom(target))
+		var/atom/AT = target
+		target_str = key_name_log(AT) + COORD(AT)
+	else
+		target_str = target
+	var/mob/MU = user
+	var/mob/MT = target
+	if(istype(MU))
+		MU.create_log(ATTACK_LOG, what_done, target, get_turf(user))
+		MU.create_attack_log("<font color='red'>Attacked [target_str]: [what_done]</font>")
+	if(istype(MT))
+		MT.create_log(DEFENSE_LOG, what_done, user, get_turf(MT))
+		MT.create_attack_log("<font color='orange'>Attacked by [user_str]: [what_done]</font>")
```

Comment:
By creating both entries, we get both the text and the datums - which is good because it allows both the full interface and quick dives into the attack logs through a simpler and more time-tested interface